app/models/display_ad.rb

```diff
@@ -35,32 +35,7 @@ class DisplayAd < ApplicationRecord
                      }
 
   def self.for_display(area, user_signed_in, article_tags = [])
-    relation = approved_and_published.where(placement_area: area).order(success_rate: :desc)
-
-    if article_tags.any?
-      display_ads_with_no_tags = relation.where(cached_tag_list: "")
-      display_ads_with_targeted_article_tags = relation.cached_tagged_with_any(article_tags)
-
-      relation = display_ads_with_no_tags.or(display_ads_with_targeted_article_tags)
-    end
-
-    if article_tags.blank?
-      relation = relation.where(cached_tag_list: "")
-    end
-
-    relation = if user_signed_in
-                 relation.where(display_to: %w[all logged_in])
-               else
-                 relation.where(display_to: %w[all logged_out])
-               end
-
-    relation.order(success_rate: :desc)
-
-    if rand(8) == 1
-      relation.sample
-    else
-      relation.limit(rand(1..15)).sample
-    end
+    DisplayAds::SampleRelevantAds.call(self, area, user_signed_in, article_tags)
```

Comment:
For this PR I'm looking for feedback on 
1. Whether we think that a service object was the right option to abstract this piece of code
2. Are there any other potential refactors that you can identify that I may have overlooked that will be useful (no matter how small).